File: binary_search/binary_search.py
#!/usr/bin/python

#That function gets sorted list and value that position we need to find.
#It returns position of value or None if list has not that value.
def binary_search(input_list, value):
    low = 0
    high = len(input_list) - 1
    
    while low <= high:
        mid = (low + high) / 2
        guess = input_list[int(mid)]
        if guess == value:
            return int(mid)
        elif guess < value:
            low = mid + 1
        else:
            high = mid - 1

    return None

mylist = [1, 2, 5, 6, 8, 10, 123]
num = int(input("Введите число для поиска "))
print(binary_search(mylist, num))

File: binary_search/binary_search.c
#include "stdio.h"

//Function gets pointer to array, number that we need to find and length of array.
//It returns number position if array has number or -1 if array hasn't
int binary_search(int array[], int num, int length) {
	int high = length - 1;
	int low = 0;
	int mid;
	while (low <= high) {
		mid = (high - low) / 2 + low;
		if (array[mid] == num) {
			return mid;
		}
		else if (array[mid] < num) {
			low = mid + 1;
		}
		else {
			high = mid - 1;
		}
	}	
	return -1;	
}

int main(int argc, char argv[]) {
	int myarray[10] = {1,2,3,4,6,8,20,64,123,5465};
	printf("Input number ");
	int number;
	scanf("%d", &number);
	int length = 10;
	printf("\nThe index is %i", binary_search(myarray, number, length));
	return 0;
}
